test/test_utils/security.py

```diff
@@ -187,3 +262,280 @@ def get_ecr_scan_allowlist_path(image_uri):
+    # In this scenario, we get error: ‘E: Could not get lock /var/lib/dpkg/lock’ while running apt-get update
+    # That is why we need multiple tries to ensure that it succeeds in one of the tries.
+    # More info: https://itsfoss.com/could-not-get-lock-error/
+    while True:
+        run_output = ctx.run(f"{docker_exec_cmd} {apt_command}", hide=True, warn=True)
+        attempt_count += 1
+        if not run_output.ok:
+            test_utils.LOGGER.info(
+                f"Attempt no. {attempt_count} on image: {image}"
+                f"Could not run apt update and upgrade. \n"
+                f"Stdout is {run_output.stdout} \n"
+                f"Stderr is {run_output.stderr} \n"
+                f"Failed status is {run_output.exited}"
+            )
+            sleep(2 * 60)
+        elif run_output.ok:
+            apt_ran_successfully_flag = True
+            break
+        if attempt_count == 10:
+            break
+    if apt_ran_successfully_flag == False:
+        raise ValueError(
+            f"Could not run apt update and upgrade on image: {image}. \n"
+            f"Stdout is {run_output.stdout} \n"
+            f"Stderr is {run_output.stderr} \n"
+            f"Failed status is {run_output.exited}"
+        )
+    ctx.run(f"docker commit {container_id} {new_image_uri}", hide=True, warn=True)
+    ctx.run(f"docker rm -f {container_id}", hide=True, warn=True)
+    ctx.run(f"docker push {new_image_uri}", hide=True, warn=True)
+
+
+def _invoke_lambda(function_name, payload_dict={}):
+    """
+    Asyncronously Invokes the passed lambda.
+
+    :param function_name: str, name of the lambda function
+    :param payload_dict: dict, payload to be sent to the lambda
+    """
+    lambda_client = boto3.client("lambda", region_name=os.getenv("REGION"))
+    response = lambda_client.invoke(
+        FunctionName=function_name, InvocationType="Event", LogType="Tail", Payload=json.dumps(payload_dict)
+    )
+    status_code = response.get("StatusCode")
+    if status_code != 202:
+        raise ValueError("Lambda call not made properly. Status code returned {status_code}")
+
+
+def create_and_save_package_list_to_s3(old_filepath, new_packages, new_filepath, s3_bucket_name):
+    """
+    This method conducts the union of packages present in the original apt-get-upgrade
+    list and new list of packages passed as an argument. It makes a new file and stores
+    the results in it.
+    :param old_filpath: str, path of original file
+    :param new_packages: list[str], consists of list of packages
+    :param new_filpath: str, path of new file that will have the results of union
+    :param s3_bucket_name: string, name of the s3 bucket
+    """
+    file1 = open(old_filepath, "r")
+    lines = file1.readlines()
+    current_packages = [line.strip() for line in lines]
+    package_list = current_packages
+    new_packages.sort()
```

Comment:
While writing the code, I was thinking that it would look better if the new packages got appended to the very end of the already existing packages. However, on reading the comment, I felt it would be better to implement the way in which you suggested.

I have made the changes as you suggested. Thanks!!